fdb-record-layer-core/src/main/proto/record_query_plan.proto

```diff
@@ -243,9 +242,16 @@ message PValue {
     PFromOrderedBytesValue from_ordered_bytes_value = 43;
     PCollateValue collate_value = 44;
     PNumericAggregationValue.PBitmapConstructAgg numeric_aggregation_value_bitmap_construct_agg = 45;
+    PMacroFunctionValue macro_function_value = 46;
   }
 }
 
+message PMacroFunctionValue {
+  optional string function_name = 1;
+  repeated PValue arguments = 2;
```

Comment:
Should this be a `repeated string` instead of a `repeated PValue`? Does using a `PValue` here mean we record type signature information, and that's what we want to preserve?

Reply:
arguments are preserved as `QuantifiedObjectValue`, which holds a `Type resultType` and a unique identifier (a uuid).